File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/lists_indexOf.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeListsIndexOf(statement: JSONObject, onError: (e: Exception) -> Unit): Int? {
    try {
        Log.d("hello", "execute_lists_indexOf")
        val pointer = statement.getJSONObject("pointer")
        val list = executeStatement(pointer["list"].toString())  as? List<Any?> ?: return null
        val item = executeStatement(pointer["item"].toString())

        return when (pointer["indexOf"].toString()) {
            "FIRST" -> list.indexOf(item) + 1
            "LAST" -> list.lastIndexOf(item) + 1
            else -> -1
        }
    } catch (e: Exception) {
        onError(e)
        return null
    }
}

File: piper/extensions/chrome/src/engine/statements/lists_indexOf.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        let pointer = statement.pointer
        let list = await this.executeStatement(stringify(pointer.list))
        let item = await this.executeStatement(stringify(pointer.item))

        switch(stringify(pointer.indexOf)) {
            case "FIRST":
                return list.indexOf(item) + 1
            case "LAST":
                return list.lastIndexOf(item) + 1
            default:
                return -1
        }
    } catch (e) {
        onError(e)
    }

    return null
}
